terraform/modules/hub/matomo.tf

```diff
@@ -0,0 +1,535 @@
+  role       = aws_iam_role.matomo_execution.name
+  policy_arn = aws_iam_policy.matomo_web_secrets.arn
+}
+
+resource "aws_cloudwatch_event_rule" "every_hour" {
+  name                = "every-hour"
+  schedule_expression = "rate(1 hour)"
+}
+
+data "template_file" "matomo_config_part_one_file" {
+  template = file("${path.module}/files/matomo/matomo-config-file-part-one.ini.php")
+
+  vars = {
+    db_host      = aws_db_instance.matomo.address
+    db_password  = random_string.matomo_db_password.result
+    trusted_host = aws_route53_record.matomo.name
+  }
+}
+
+resource "aws_ssm_parameter" "matomo_config_file_part_one" {
+  name        = "/matomo/config-file-part-one"
+  description = "Base64 encoded config file part one for Matomo"
+  type        = "SecureString"
+  key_id      = aws_kms_key.matomo.arn
+  value       = base64encode(data.template_file.matomo_config_part_one_file.rendered)
+}
+
+resource "aws_ssm_parameter" "matomo_config_file_part_two" {
+  name        = "/matomo/config-file-part-two"
+  description = "Base64 encoded config file part two for Matomo"
+  type        = "SecureString"
+  key_id      = aws_kms_key.matomo.arn
+  value       = base64encode(file("${path.module}/files/matomo/matomo-config-file-part-two.ini.php"))
+}
+
+resource "random_string" "matomo_db_password" {
+  length  = 41
+  special = false
+}
+
+resource "aws_ssm_parameter" "matomo_db_password" {
+  name   = "/matomo/mysql-password"
+  type   = "SecureString"
+  key_id = aws_kms_key.matomo.arn
+  value  = random_string.matomo_db_password.result
+}
+
+resource "aws_db_subnet_group" "matomo_db" {
+  name       = "matomo-db"
+  subnet_ids = aws_subnet.internal.*.id
+}
+
+locals {
+  engine         = "mysql"
+  engine_version = "5.7"
+}
+
+resource "aws_db_parameter_group" "matomo" {
+  name   = "matamo-rds"
+  family = "${local.engine}${local.engine_version}"
+
+  parameter {
+    name  = "max_allowed_packet"
+    value = "67108864"
+  }
+}
+
+resource "aws_db_instance" "matomo" {
+  allocated_storage         = 500
+  engine                    = local.engine
+  engine_version            = local.engine_version
+  storage_type              = "gp2"
+  identifier                = "matomo"
+  name                      = "matomo"
+  username                  = "matomo"
+  password                  = random_string.matomo_db_password.result
+  vpc_security_group_ids    = [aws_security_group.matomo_db.id]
+  db_subnet_group_name      = aws_db_subnet_group.matomo_db.name
+  final_snapshot_identifier = "matomo-final"
+  backup_retention_period   = 7
+  storage_encrypted         = true
+  parameter_group_name      = aws_db_parameter_group.matomo.name
+  deletion_protection       = true
+
+  ca_cert_identifier = "rds-ca-2019"
+
+  # 8x vCPU, 25.5x ECU, 32GiB RAM
+  # more here: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.DBInstanceClass.html
+  instance_class = "db.m4.2xlarge"
```

Comment:
We don't need db.m4.2xlarge for non-production environments.

Reply:
We still don't need db.m4.2xlarge for non-production environments. We should replace it with a variable and configure it to use an appropriate instance class for each environment.